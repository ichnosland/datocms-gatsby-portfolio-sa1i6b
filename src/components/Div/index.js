/**
*
* Div
*
*/

import styled, { css } from 'styled-components';
import media from 'style/mediainjector';

const Div = styled.div`
  display: ${(props) => props.display ? props.display : 'block'};
  position: ${(props) => props.position ? props.position : 'relative'};
  width: ${(props) => props.width};
  max-width: ${(props) => props.maxWidth ? props.maxWidth : '100%'};
  height: ${(props) => props.height};
  color: ${(props) => props.color ? props.color : 'inherit'};
  text-align: ${(props) => props.align};
  float: ${(props) => props.float ? props.float : 'none'};
  margin: ${(props) => props.margin ? props.margin : '0'};
  padding: ${(props) => props.padding ? props.padding : '0'};
  border: ${(props) => props.border ? props.border : 'none'};
  border-width: ${(props) => props.borderwidth};
  border-style: ${(props) => props.borderstyle};
  border-color: ${(props) => props.bordercolor};
  background-color: ${(props) => props.bgColor ? props.bgColor : 'transparent'};
  border-radius: ${(props) => props.radius};
  overflow: ${(props) => props.overflow ? props.overflow : 'visible'};
  user-select: ${(props) => props.userSelect};
  pointer-events: ${(props) => props.pointerEvents};
  z-index: ${(props) => props.zindex};
  ${(props) => props.nobreak && css`
    page-break-inside: avoid;
  `}
  ${media.print`
    ${({ noPrint }) => noPrint && css`
        display:none;
    `}
  `}
`;

export default Div;
