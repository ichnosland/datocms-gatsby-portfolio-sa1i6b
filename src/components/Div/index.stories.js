import React from 'react';
import { storiesOf } from '@storybook/react';
import { withNotes } from '@storybook/addon-notes';
import { withKnobs, text, boolean, color } from '@storybook/addon-knobs';
import Div from './index';


const nota = 'La props nobreak assegna \'page-break-inside: avoid;\' \n quella noPrint \'display:none;\' in stampa';

storiesOf('Div', module)
  .addDecorator(withKnobs)
  .addWithJSX('Div', withNotes(nota)(() => (
    <Div
      display={text('display', 'block')}
      position={text('position', 'relative')}
      width={text('width', 'auto')}
      maxWidth={text('maxWidth', '100%')}
      height={text('height', '')}
      color={color('color', 'inherit')}
      align={text('align (tex-align)', '')}
      float={text('float', 'none')}
      margin={text('margin', '0')}
      padding={text('padding', '0')}
      radius={text('radius', '')}
      bgColor={color('bgColor', 'transparent')}
      border={text('border', '')}
      borderwidth={text('borderwidth', '')}
      borderstyle={text('borderstyle', '')}
      bordercolor={color('bordercolor', '')}
      overflow={text('overflow', 'visible')}
      userSelect={text('userSelect', 'auto')}
      pointerEvents={text('pointerEvents', 'auto')}
      zindex={text('zindex', 'auto')}
      nobreak={boolean('nobreak', false)}
      noPrint={boolean('noPrint', true)}
    >
      <p>Ciao, sono all&apos;interno di un Div. Usa le props per personalizzarlo :)</p>
    </Div>
  )));
