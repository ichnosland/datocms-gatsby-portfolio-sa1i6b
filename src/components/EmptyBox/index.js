/**
*
* EmptyBox
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Div from '../Div';
import folder from 'images/empty-folder.png';
import { colore } from 'style/color';

export const Folder = styled.img`
  width: 30%;
  padding: 40px 0;
`;

class EmptyBox extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <Div color={colore.ui.lightTxt} align="center">
        <Folder src={folder} />
        <p>{this.props.text}</p>
      </Div>
    );
  }
}

EmptyBox.propTypes = {
  text: PropTypes.string,
};

export default EmptyBox;
