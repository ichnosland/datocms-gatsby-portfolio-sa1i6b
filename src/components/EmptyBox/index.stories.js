import React from 'react';
import { storiesOf } from '@storybook/react';
import { withNotes } from '@storybook/addon-notes';
import { withKnobs, text, select } from '@storybook/addon-knobs';
import { ThemeProvider } from 'styled-components';
import { allThemes, themeOptions } from 'style/theme';
import EmptyBox from './index';

const nota = `
  Per ora c'è solo la versione con il folder vuoto.\n
  Più avanti inseriremo altre immagini/opzioni e renderemo l'src del'img una props.\n
  Suggerisco anche di rinominare il comp in 'EmptyState', mi pare più esplicito.
`;

storiesOf('EmptyBox', module)
  .addDecorator(withKnobs)
  .add('EmptyBox', withNotes(nota)(() => (
    <ThemeProvider theme={allThemes[select('theme', themeOptions, 'alatin')]}>
      <EmptyBox
        text={text('text', 'Testo informativo')}
      />
    </ThemeProvider>
  )));

