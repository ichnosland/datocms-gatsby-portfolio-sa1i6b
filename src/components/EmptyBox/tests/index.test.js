import React from 'react';
import { shallow } from 'enzyme';

import Div from '../Div';
import folder from 'images/empty-folder.png';
import EmptyBox, { Folder } from '../index';

describe('<EmptyBox />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(
      <EmptyBox />
    );
    expect(renderedComponent.find(Div).length).toBe(1);
  });

  it('should render text w/ props text', () => {
    const text = 'Questo Ã¨ un testo';
    const renderedComponent = shallow(
      <EmptyBox text={text} />
    );
    expect(renderedComponent.find(Div).length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<Folder />', () => {
  it('should render an <img>', () => {
    const src = folder;
    const renderedComponent = shallow(
      <Folder src={src} />
    );
    expect(renderedComponent.type()).toEqual('img');
    expect(renderedComponent).toMatchSnapshot();
  });
});
