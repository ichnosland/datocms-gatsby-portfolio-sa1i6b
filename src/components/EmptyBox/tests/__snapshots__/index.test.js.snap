// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyBox /> should render text w/ props text 1`] = `
<Div
  align="center"
  color="rgb(164, 164, 164)"
>
  <EmptyBox__Folder
    src="IMAGE_MOCK"
  />
  <p>
    Questo Ã¨ un testo
  </p>
</Div>
`;

exports[`<Folder /> should render an <img> 1`] = `
.c0 {
  width: 30%;
  padding: 40px 0;
}

<img
  className="c0"
  src="IMAGE_MOCK"
/>
`;
